lib/assert.js

```diff
@@ -91,6 +91,36 @@ export function isArrayOfClass(object, expectedClass) {
   }
 }
 
+export function isInterface(object, expectedProperties) {
+  const parameter = Object.keys(object).pop();
+  const message = `The '${parameter}' parameter should be an object that has
```

Comment:
Should you be creating this inside the for loop to update the message for the failing property?